lib/test.js

```diff
@@ -120,8 +120,27 @@ function runMocha(state, callback) {
   global.assertNoError = err => {
     if(err) {
       const obj = err instanceof BedrockError ? err.toObject() : err;
+      // only one cache for each error.
+      const errorCache = new Set();
+      // JSON.stringify will call this for each key in the json
+      // we need to catch circular references here.
+      const removeCircularReferences = (key, value) => {
+        // javascript passes non-object primitives by value
+        // however objects (including arrays) are by reference
+        if(typeof value === 'object' && value !== null) {
+          // if the value has already been cached it's a circular reference
+          if(errorCache.has(value)) {
+            return;
+          }
+          // otherwise cache it so we catch any future circular references
+          errorCache.add(value);
+        }
+        return value;
+      }
       console.error(
-        'An unexpected error occurred:', JSON.stringify(obj, null, 2));
+        'An unexpected error occurred:',
+        JSON.stringify(obj, removeCircularReferences, 2)
```

Comment:
Was this using `console.error` for some specific reason?  Maybe too low level at this point?  Looks like a `logger` instance is available so maybe `logger.error` could be used instead?  That code already [handles cycles](https://github.com/digitalbazaar/bedrock/blob/master/lib/loggers.js#L289-L294).  If `console.error` is needed here, might be better to just copy the `cycle` code instead of rolling something new.

Reply:
@mattcollier switch to bedrock-logger perhaps or is there so the test still receives the error?

Reply:
the purpose of this helper is to display error in a mocha test suite on the console.  I don't think we want to change this core functionality.

Reply:
if using a bedrock logger results in errors on the console and allows for code re-use, that's coo.

Reply:
Sometimes in these specialized cases using `console.error` is fine too.  If that's used, I think the `cycle.decycle` usage is better than maintaining custom code.

Reply:
@davidlehn @mattcollier I tested using the bedrock logger and it does not log during the mocha test run. so maybe we should re-use `const cycle = require('cycle');` here.

Reply:
@mattcollier https://github.com/digitalbazaar/bedrock-test/pull/13/commits/a89b0f04a1b2285599601c0800617f2d3b756981

should solve it. @davidlehn is ok with this being merged.